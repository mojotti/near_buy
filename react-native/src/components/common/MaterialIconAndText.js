import React from 'react';
import { Dimensions, Text, View } from 'react-native';
import PropTypes from 'prop-types';
import MaterialIcon from 'react-native-vector-icons/MaterialCommunityIcons';
import { styles } from '../../static/styles/MaterialIconAndTextStyles';

const { width } = Dimensions.get('window');


export default class MaterialIconAndText extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.iconContainer}>
          <MaterialIcon
            name={this.props.iconName}
            size={width * 0.22}
            style={styles.iconStyles}
          />
        </View>
        <Text style={styles.noItemsText}>{this.props.headerText}</Text>
        <Text style={styles.noItemsText}>{this.props.containerText}</Text>
      </View>
    );
  }
}

MaterialIconAndText.propTypes = {
  iconName: PropTypes.string.isRequired,
  headerText: PropTypes.string.isRequired,
  containerText: PropTypes.string.isRequired,
};
