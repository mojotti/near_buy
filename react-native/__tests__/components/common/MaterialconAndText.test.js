import React from 'react';
import { shallow } from 'enzyme';
import MaterialIconAndText from '../../../src/components/common/MaterialIconAndText';


describe('<MaterialIconAndText />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <MaterialIconAndText
        iconName={'access-point'}
        headerText={'this is header'}
        containerText={'and this just context'}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
