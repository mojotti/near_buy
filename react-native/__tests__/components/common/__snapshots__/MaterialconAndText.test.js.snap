// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialIconAndText /> renders correctly 1`] = `
<Component
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 100,
        "height": 180,
        "marginBottom": 15,
        "overflow": "hidden",
        "width": 180,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#727272"
      name="access-point"
      size={165}
      style={
        Object {
          "alignSelf": "center",
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "color": "#727272",
        "fontFamily": null,
        "paddingHorizontal": 30,
        "textAlign": "center",
      }
    }
  >
    this is header
  </Component>
  <Component
    style={
      Object {
        "color": "#727272",
        "fontFamily": null,
        "paddingHorizontal": 30,
        "textAlign": "center",
      }
    }
  >
    and this just context
  </Component>
</Component>
`;
